### PR TITLE
New producer to update JEC and Userdata of pat::Jets

### PR DESCRIPTION
**Pull request by ahinzmann**
This module takes pat::Jets as input, updates JEC and Userdata information and writes out new pat::Jets.
The use case is to run on MiniAOD samples with outdated JEC and missing Userdata.

The test configuration "PhysicsTools/PatAlgos/test/update_jets_from_MiniAOD.py" tests boths features.
patJetsUpdated gives the jets with JEC from the given global tag and additional user data.
patJetsUndoJEC removes JEC from patJetUpdated
patJetsReapplyJEC reapplies JEC on patJetsUndoJEC

### CODE REVIEW

**Comment by ferencek** [comment]
This is a minor detail but more descriptive jet collection names could be helpful. For instance

`patJetsUpdated` --> `patJetsUpdatedJEC`
`patJetsUpdated2` --> `patJetsUncorr`
`patJetsUpdated3` --> `patJetsReappliedJEC`

Just a suggestion.

**Comment by monttj** [comment]
+1

File: PhysicsTools/PatAlgos/plugins/PATJetUpdater.cc [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+using namespace pat;
+
+
+PATJetUpdater::PATJetUpdater(const edm::ParameterSet& iConfig) :
+  useUserData_(iConfig.exists("userData"))
+{
+  // initialize configurables
+  jetsToken_ = consumes<edm::View<Jet> >(iConfig.getParameter<edm::InputTag>( "jetSource" ));
+  addJetCorrFactors_ = iConfig.getParameter<bool>( "addJetCorrFactors" );
+  jetCorrFactorsTokens_ = edm::vector_transform(iConfig.getParameter<std::vector<edm::InputTag> >( "jetCorrFactorsSource" ), [this](edm::InputTag const & tag){return mayConsume<edm::ValueMap<JetCorrFactors> >(tag);});
```
Comment (davidlange6) [inline PhysicsTools/PatAlgos/plugins/PATJetUpdater.cc]: whether or not the ValueMap<JetCorrFactors> will be consumed is known according to the value of addJetCorrFactors_. Correct? if so, change the mayConsume to consume (within an if statement on the value of addJetCorrFactors_)

**Comment by ahinzmann** [comment]
This got forward ported from 7_4_X #8288 already.

**Pull request closed without merging**